server/gitrest/packages/gitrest-base/src/utils/gitWholeSummaryManager.ts

```diff
@@ -255,15 +377,76 @@ export class GitWholeSummaryManager {
         }
 
         return {
-            isNew,
+            isNew: isNewDocument,
             writeSummaryResponse: {
                 id: commit.sha,
             },
         };
     }
 
+    private async writeSummaryTree(
+        wholeSummaryTreeEntries: WholeSummaryTreeEntry[],
+        useLowIoWrite: boolean = false,
+    ): Promise<string> {
+        if (!useLowIoWrite) {
+            return this.writeSummaryTreeCore(
+                wholeSummaryTreeEntries,
+                this.repoManager,
+            );
+        }
+
+        Lumberjack.info("Low-IO mode: Initializing In-memory Filesystem");
+        const inMemoryFsManagerFactory = new MemFsManagerFactory();
+        const inMemoryRepoManagerFactory = new IsomorphicGitManagerFactory(
+            {
+                baseDir: "/usr/gitrest",
+                useRepoOwner: true,
+            },
+            inMemoryFsManagerFactory,
+            new NullExternalStorageManager(),
+            true, /* repoPerDocEnabled */
+            false, /* enableRepositoryManagerMetrics */
+        );
+        const inMemoryRepoManager = await inMemoryRepoManagerFactory.create(
+            {
+                repoOwner: "gitrest",
+                repoName: this.documentId,
+                storageRoutingId: {
+                    tenantId: "internal",
+                    documentId: this.documentId,
+                },
+            }
+        );
+        Lumberjack.info("Low-IO mode: Writing summary to memory");
+        const inMemorySummaryTreeHandle = await this.writeSummaryTreeCore(
+            wholeSummaryTreeEntries,
+            inMemoryRepoManager,
+        );
+        Lumberjack.info("Low-IO mode: Reading summary from memory");
+        const gitTree = await inMemoryRepoManager.getTree(inMemorySummaryTreeHandle, true /* recursive */);
+        const fullGitTree = await buildFullGitTreeFromGitTree(
+            gitTree,
+            inMemoryRepoManager,
+        );
+        Lumberjack.info("Low-IO mode: Writing summary to storage");
+        const summaryTreeHandle = this.writeSummaryTreeCore(
+            [{
```

Comment:
Generate a metric to capture tenant id, doc id, content length, and processing time?

Reply:
Or if we want to keep this as a long, we can also add those properties in the properties parameter for Lumberjack.info().

Reply:
I removed most of these inner logs, but added additional properties to the wrapping summary write metric that will help us understand each one in our telemetry (newDocument, commit/tree sha, and enableLowIoWrite).